pkg/sql/logictest/testdata/logic_test/mixed_version_can_login

```diff
@@ -23,7 +23,7 @@ upgrade 1
 query B
 SELECT crdb_internal.node_executable_version() SIMILAR TO '1000023.1-%'
 ----
-true
+false
```

Comment:
we shouldn't just rewrite these to false. instead, change the assertion so it's `SIMILAR TO 000023.2-%`